src/SIPSorceryMedia.FFmpeg.csproj

```diff
@@ -13,17 +13,19 @@
  
   <ItemGroup>
     <PackageReference Include="DirectShowLib.Standard" Version="2.1.0" />
+    <PackageReference Include="FFmpeg.AutoGen.Bindings.DynamicallyLoaded" Version="6.0.0" />

```

Comment:
Whats the benefit of switching from FFmpeg.AutoGen to FFmpeg.AutoGen.Bindings.DynamicallyLoaded?

Reply:
If we don't use FFmpeg.AutoGen.Bindings.DynamicallyLoaded a 'PlatformNotSupportedException' will be thrown at: https://github.com/Ruslan-B/FFmpeg.AutoGen/blob/15d15bdb5230fb1457883ad2b918cb94e6e1be59/FFmpeg.AutoGen/FFmpeg.cs#L17
More about that in this [issue](https://github.com/Ruslan-B/FFmpeg.AutoGen/issues/260)

Reply:
Ok but that issue is specifically for Android. How is using the dynamically loaded approach going to affect all the existing people that use this library on Winwows, macOS and Linux? Perhahps it would be better to use a precompiler directive for Android and only use the dynamic approach with it until it's clear whether there are any side effects for other OS'es.

Reply:
The only difference I can see is at this line: https://github.com/Ruslan-B/FFmpeg.AutoGen/blob/e129a3128b570e26f4c934cecbb9c00bab3c5dc4/FFmpeg.AutoGen.Abstractions/FFmpeg.cs#L23, it uses `Environment.OsVersion.Platform` instead of `FunctionResolverFactory.GetPlatformId` that throws the `PlatformNotSupportedException`.
otherwise the FunctionResolvers is the same and the bindings is auto generated.
you can know more about why is the FFmpeg.AutoGen.Abstractions and FFmpeg.AutoGen.Bindings.DynamicallyLoaded are created [here](https://github.com/Ruslan-B/FFmpeg.AutoGen/commit/534a3eb0ff295bc9bc8382b311df407180ab5051) and [here](https://github.com/Ruslan-B/FFmpeg.AutoGen/issues/210#issuecomment-1209805082)
I have tried this library with the FFmpeg.AutoGen.Bindings.DynamicallyLoaded on Windows and it worked well.
Also, we can't use a precompiler directive for Android because some struct names are different for example `byte_array4` is `byte4` making a precompiler directive for all of these will be big hassle.

Reply:
Ok sounds reasonable. I'd be happy to merge and do some testing myself prior to any nuget release.

@ChristopheI do you see any red flags?